File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline


if __name__ == '__main__':
    stack = []
    res = []
    N = int(input())
    for _ in range(N):
        order = input().split()
        if order[0] == 'push':
            stack.append(order[1])
        elif order[0] == 'pop':
            res.append(stack.pop() if stack else -1)
        elif order[0] == 'size':
            res.append(len(stack))
        elif order[0] == 'empty':
            res.append(0 if stack else 1)
        elif order[0] == 'top':
            res.append(stack[-1] if stack else -1)

    for r in res:
        print(r)

File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <stack>
#include <string>

using namespace std;

int main() {
    stack<int> Q;
    
    int N;
    cin >> N;
    
    string order;
    for (int i = 0; i < N; i++) {
        cin >> order;
        if (order == "push") {
            int x;
            cin >> x;
            Q.push(x);
        } else if (order == "pop") {
            if (Q.empty()) {
                cout << -1 << endl;
            } else {
                cout << Q.top() << endl;
                Q.pop();
            }
        } else if (order == "size") {
            cout << Q.size() << endl;
        } else if (order == "empty") {
            cout << (Q.empty()) << endl;
        } else if (order == "top") {
            if (Q.empty()) {
                cout << -1 << endl;
            } else {
                cout << Q.top() << endl;
            }
        }
    }

    return 0;
}
